tests/Aspire.Hosting.MySql.Tests/MySqlFunctionalTests.cs

```diff
@@ -0,0 +1,406 @@
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+                    {
+                        [$"ConnectionStrings:{db2.Resource.Name}"] = await db2.Resource.ConnectionStringExpression.GetValueAsync(default)
+                    });
+
+                    hb.AddMySqlDataSource(db2.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        // Wait until the database is available
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            using var connection = host.Services.GetRequiredService<MySqlConnection>();
+                            await connection.OpenAsync(token);
+                            Assert.Equal(ConnectionState.Open, connection.State);
+                        }, cts.Token);
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            using var connection = host.Services.GetRequiredService<MySqlConnection>();
+                            await connection.OpenAsync(token);
+
+                            var command = connection.CreateCommand();
+                            command.CommandText = $"SELECT * FROM cars;";
+                            var results = await command.ExecuteReaderAsync(token);
+
+                            Assert.True(results.HasRows);
+                        }, cts.Token);
+                    }
+
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+
+        }
+        finally
+        {
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    Directory.Delete(bindMountPath);
+                }
+                catch
+                {
+                    // Don't fail test if we can't clean the temporary folder
+                }
+            }
+        }
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyWithInitBindMount()
+    {
+        // Creates a script that should be executed when the container is initialized.
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, BackoffType = DelayBackoffType.Linear, Delay = TimeSpan.FromSeconds(2), ShouldHandle = new PredicateBuilder().Handle<MySqlException>() })
+            .Build();
+
+        var bindMountPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
```

Comment:
Would it be better to use `Directory.CreateTempSubdirectory().FullName` here too, and skip the need to delete+create the directory below?